### PR TITLE
Convert EDITOR_ISSUE_TEMPLATE from markdown to YAML

### LINKED ISSUE #2625: new editor issue template

### PR DESCRIPTION
**Pull request by RaoOfPhysics**
Signed-off-by: Achintya Rao <achintya@achintyarao.in>

### Summary

Fixes #2625 

@aldenc and I are updating the templates from Markdown to YAML. Here’s the first one!

### List of changes proposed in this PR (pull-request)

* The basic set of changes is simply converting existing content.
* A second set of minor changes involves adding a checklist at the end for the updates.

### What should a reviewer concentrate their feedback on?

- [ ] Does the flow make sense?

### Acknowledging contributors

- [x] All contributors to this pull request are already named in the [table of contributors](https://github.com/alan-turing-institute/the-turing-way/blob/main/README.md#contributors) in the README file.

### CODE REVIEW

**Comment by RaoOfPhysics** [comment]
I did a `git mv` to rename the file but for some reason it deleted the old one and created a new one. :(

**Comment by sgibson91** [comment]
> I did a `git mv` to rename the file but for some reason it deleted the old one and created a new one. :(

Because the *contents* changed as well. If you had just don't a `git mv`, not edited anything, committed that and opened a PR, then you would see "file renamed without changes". But as soon as your rename a file and then edit the contents, then it looks like a "delete one file, create a new fie" situation. This is to do with git's object database and how it tracks this info.

**Comment by RaoOfPhysics** [comment]
Ah, thanks, Sarah! So I should treat editing and renaming as separate commits in the future.

(Also, after a lot of wrestling with the YAML, I think I got it to work. And have force-pushed loads of re-writes so there’s only one commit in the end…)

File: .github/ISSUE_TEMPLATE/EDITOR_ISSUE_TEMPLATE.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    label: Summary
+    description: "Please provide a detailed description of the chapter/subchapter you propose to edit, review or restructure. Please provide as much context as possible and link to related issues and/or pull requests. Providing links to sections you're referencing for clarity is highly encouraged."
+  validations:
+    required: true
+- type: textarea
+  id: todo
+  attributes:
+    label: "What needs to be done?"
+    description: "Please comment on the overall structure of the file. Your response can cover the questions such as, “Do sections need clearer headings? Are there any technical terms that might not be understood? Is something missing that is worth discussing before adding?” We suggest using bullets (indicated by * or -) and filled checkboxes ([x]) here."
+    placeholder: Please use bullets (* or -) and filled checkboxes ([x]) as necessary.
```
Comment (sgibson91) [inline .github/ISSUE_TEMPLATE/EDITOR_ISSUE_TEMPLATE.yml:20]:
```suggestion
    description: "Please comment on the overall structure of the file. Your response can cover the questions such as, “Do sections need clearer headings? Are there any technical terms that might not be understood? Is something missing that is worth discussing before adding?”"
    placeholder: Please use bullets (* or -) and filled checkboxes ([x]) as necessary.
```

My opinion is that it doesn't add much value to repeat this in both the description and the placeholder

**Review by sgibson91 — APPROVED** [review]
I have one small opinion-style comment but if it doesn't resonate with you, please merge without it :)

**Comment by sgibson91** [comment]
> So I should treat editing and renaming as separate commits in the future

Sorry, I don't think I was completely clear. As soon as you edit a renamed file within the same PR, it will look like you've deleted one file and added a new one. Even if your process was: rename a file, commit, edit the file, commit. This is because information in the git object database will have changed. It will only be a clear "file renamed without changes" if that was the *only* change you made to that file in the whole PR.

**Comment by RaoOfPhysics** [comment]
Ah, I see. I’ve only ever used it while working with myself in the past, so haven’t encountered this issue before.

**Comment by RaoOfPhysics** [comment]
Thanks for the review, @sgibson91. Now merging!

**Pull request merged**